src/extension/highlighedMatches.ts

```diff
@@ -0,0 +1,134 @@
+/*
+This file handles getting highlighted matches from different services
+ */
+import axios, { AxiosResponse } from 'axios';
+import { UnhandledListenForCb } from 'nodecg/lib/nodecg-instance';
+import * as nodecgContext from './util/nodecg';
+import { HighlightedMatch } from 'schemas';
+import { Team } from 'types/team';
+import { BattlefyStage, Match } from './types/battlefyStages';
+
+const nodecg = nodecgContext.get();
+
+const highlightedMatchData = nodecg.Replicant<HighlightedMatch>('highlighedMatches');
+
+/**
+ * Handles the listener for when request is triggered
+ */
+nodecg.listenFor('getHighlightedMatches', async (data, ack: UnhandledListenForCb) => {
+    if (!data.stages || !data.provider) {
+        ack(new Error('Missing arguments.'), null);
+        return;
+    }
+    switch (data.provider) {
+        case 'Battlefy':
+            getBattlefyMatches(data.stages)
+                .then(data => {
+                    updateMatchesReplicants(data);
+                    ack(null, data);
+                })
+                .catch(err => {
+                    ack(err);
+                });
+            break;
+        default:
+            ack(new Error('Invalid method given.'));
+    }
+});
+
+/**
+ * Assigns data to highlighedMatch replicant
+ * @param data Data handed back from data provider
+ */
+export function updateMatchesReplicants(data: HighlightedMatch): void {
+    // Only assign the data to replicant if there is data
+    if (data.length > 0) {
+        highlightedMatchData.value = data;
+    }
+}
+
+/**
+ * Returns the matches with isMarkedLive as true for a list of Battlefy Stages
+ * @param stages StageID of the stages to get highlighted matches from
+ */
+async function getBattlefyMatches(stages: Array<string>): Promise<HighlightedMatch> {
+    const matchData: HighlightedMatch = [];
+    const requestPromises: Promise<AxiosResponse>[] = [];
+    for (let i = 0; i < stages.length; i++) {
```

Comment:
Looking into it, it's possible to get matches for every stage in a tournament: try querying `GET https://dtmwra1jsgyb0.cloudfront.net/tournaments/609197f9cf99e72a16a97d65?extend[stages][$query][deletedAt][$exists]=false&extend[stages][$opts][name]=1&extend[stages][$opts][bracket]=1&extend[stages][matches]`
Maybe it'd be easier to just always pull every highlighted match for a tournament instead of querying either "All stages" or one specific stage?

Reply:
I query the way I do due to it being how the battlefy Website does it, better staying with it as we know it's supported method

Reply:
Most of the methods we use to query Battlefy are unofficial and technically unsupported, and the way this is written is hardly an exception. I think that if we can do one query to Battlefy instead of multiple that'd be a worthwhile thing to do